src/components/subgoal/Subgoal-Assignment-Modal.tsx

```diff
@@ -0,0 +1,91 @@
+import { trpc } from "@/client/lib/trpc";
+import {
+  Box,
+  Dialog,
+  DialogTitle,
+  Alert,
+  AlertTitle,
+  Button,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  Checkbox,
+  ListItemText,
+} from "@mui/material";
+import { useState } from "react";
+
+interface SubgoalAssignmentModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const SubgoalAssignmentModal = (props: SubgoalAssignmentModalProps) => {
+  const [selectedParaIds, setSelectedParaIds] = useState<string[]>([]);
+  const myParasQuery = trpc.case_manager.getMyParas.useQuery();
+
+  const handleParaToggle = (paraId: string) => () => {
+    setSelectedParaIds((prev) => {
+      if (prev.includes(paraId)) {
+        return prev.filter((id) => id !== paraId);
+      } else {
+        return [...prev, paraId];
+      }
+    });
+  };
+
+  const handleClose = () => {
+    props.onClose();
+    setSelectedParaIds([]);
+  };
+
+  return (
+    <Dialog open={props.isOpen} onClose={handleClose}>
+      <DialogTitle>Assign to benchmark</DialogTitle>
+
+      <Box sx={{ px: 2, pb: 2 }}>
+        <Alert severity="info" sx={{ mb: 2 }}>
+          <AlertTitle>Benchmark</AlertTitle>
+          By October, when given a list of 20 unfamiliar words that contain
```

Comment:
where can we pull this from?

Reply:
Looks like the closest API we've made is iep.getSubgoalAndTrialData, for data collection, which we are using in /src/pages/benchmarks/[benchmark_id]/index.tsx, but we do not yet have an API for an individual Subgoal without the Trial data.

Reply:
We also have iep.tempAddTaskToSelf for assignment,being used in the Subgoal.tsx